gateway/src/apicast/oauth/oidc.lua

```diff
@@ -185,8 +185,12 @@ function _M:verify(jwt, cache_key)
   -- Find jwk with matching kid for current JWT in request
   local jwk_obj = find_jwk(jwt, self.keys)
 
+  if jwk_obj == nil then
+    return false, '[jwk] jwk not found, token might not belong to right realm'
```

Comment:
I would suggest the following log message which would be more explicit and accurate:
```
ngx.log(ngx.ERR, "[jwt] failed verification for kid: ", jwt.header.kid)
return false, '[jwk] not found, token might belong to a different realm'
```

This would help customers identify the issue quicker and without having to enable debug log level.